### PR TITLE
[pnnl/gmt#9] C++ porting

### PR DESCRIPTION
**Pull request by mdrocco**
compiles with g++ 7.3, not tested yet

### CODE REVIEW

**Comment by mminutoli** [comment]
@droccom I will wait for the patch set to be tested before reviewing it.

**Comment by mdrocco** [comment]
All the tests passing from the C version also pass from the C++ version.

**Comment by mdrocco** [comment]
@mminutoli All the tests pass also with multiple nodes, I think you can proceed and review the PR.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,9 @@ set(sources
       comm_server.c  gmt_execute.c  gmt_misc.c    gmt_ucontext.c  memory.c  profiling.c  utils.c
       config.c       gmt_for.c      helper.c      mtask.c   timing.c     worker.c
 )
+set_source_files_properties(${sources} PROPERTIES LANGUAGE CXX )
 
 add_library(gmt ${sources})
-target_link_libraries(gmt ${MPI_LIBRARIES})
+set_target_properties(gmt PROPERTIES LINKER_LANGUAGE CXX)
+target_link_libraries(gmt ${MPI_LIBRARIES} pthread)
```
Comment (mminutoli) [inline src/CMakeLists.txt]: The build system should be refactored, but that's a task on its own.  Please, change the relevant part into something like this:
```
find_package(Threads REQUIRED)
include_directories(${THREADS_PTHREADS_INCLUDE_DIR})

add_library(gmt ${sources})
target_link_libraries(gmt ${MPI_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
```

**Review by mminutoli — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -65,11 +65,9 @@ link_libraries(m)
 include_directories(${GMT_MAIN_INCLUDE_DIR} ${GMT_INCLUDE_DIR})
 
 
-# Select the appropriate GCC language standard semantics for "external inline"
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fgnu89-inline -std=gnu99")
 # Appending -O2 to gmt because otherwise it will fail to link.
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -pthread")
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wabi -Wextra")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -pthread")
```
Comment (mminutoli) [inline CMakeLists.txt]: Remove -pthread here and use the flag from the comment below.

**Comment by mdrocco** [comment]
commit 427954a4eb32d897eddc1c121a141730eb3b846b should clear the review

**Review by mminutoli — APPROVED** [review]

**Pull request merged**